source/common/http/codec_client.cc

```diff
@@ -37,13 +41,17 @@ void CodecClient::deleteRequest(ActiveRequest& request) {
   if (codec_client_callbacks_) {
     codec_client_callbacks_->onStreamDestroy();
   }
+  if (numActiveRequests() == 0) {
+    enableIdleTimer();
```

Comment:
Should this also be enabled initially? Or do we assume we will always have an inflight request at codec creation?

Reply:
I went with that assumption and in tests that I did it is true also. I added that condition to ensure that it is enabled only when there are no active streams.